nox/_parametrize.py

```diff
@@ -78,10 +91,14 @@ def __eq__(self, other: object) -> bool:
         raise NotImplementedError
 
 
-def _apply_param_specs(param_specs: List[Param], f: Any) -> Any:
+def _apply_param_specs(
+    param_specs: List[Param], sessionparams: AbstractSet[str], f: Any
+) -> Any:
     previous_param_specs = getattr(f, "parametrize", None)
     new_param_specs = update_param_specs(previous_param_specs, param_specs)
+    sessionparams |= getattr(f, "sessionparams", frozenset())
```

Comment:
Just curious: is this equivalent to 

```suggestion
    sessionparams.update(getattr(f, "sessionparams", frozenset()))
```

? If so readability might suggest the latter but this is maybe because I am an old pythonista already :)

Reply:
Join the club :wink: Yes, the two are almost synonymous, except that `update` accepts any iterable; both have been around since [PEP 218] introduced sets. I'm happy to change this if it's confusing. Personally, I find that using set union and assignment instead of a single mutating function call makes the code more declarative and idiomatic, and avoiding parentheses improves readability.

[PEP 218]: https://www.python.org/dev/peps/pep-0218/

Reply:
I learnt something today thanks ! :) Let's let the core nox team decide then.